src/components/HTMLEngineProvider/HTMLRenderers/index.ts

```diff
@@ -10,17 +11,21 @@ import PreRenderer from './PreRenderer';
 /**
  * This collection defines our custom renderers. It is a mapping from HTML tag type to the corresponding component.
  */
-export default {
+const HTMLEngineProviderComponentList: CustomTagRendererRecord = {
     // Standard HTML tag renderers
     a: AnchorRenderer,
-    code: CodeRenderer,
+    code: CodeRenderer as CustomTextualRenderer,
     img: ImageRenderer,
     video: AnchorRenderer, // temporary until we have a video player component
 
     // Custom tag renderers
     edited: EditedRenderer,
     pre: PreRenderer,
-    'mention-user': MentionUserRenderer,
-    'mention-here': MentionHereRenderer,
-    'next-step-email': NextStepEmailRenderer,
+    /* eslint-disable @typescript-eslint/naming-convention */
+    'mention-user': MentionUserRenderer as CustomTextualRenderer,
+    'mention-here': MentionHereRenderer as CustomTextualRenderer,
```

Comment:
```suggestion
    'mention-here': MentionHereRenderer,
```
You can get rid of assertion updating `MentionHereRenderer` typing to use `CustomRendererProps<TText | TPhrasing>` instead  of just `CustomRendererProps<TText>`. Same for every place with assertion in this file
```
function MentionHereRenderer({style, tnode}: CustomRendererProps<TText | TPhrasing>) {
```